@VincentLanglet Thank you! Works for SF 4.4. But with SF 6.2 I have other issues now.

Below output is from the SF 4.4 project, Symfony 6.2 follows later:

Below:
```bash
$ vendor/bin/phpstan analyse src -c phpstan.neon --level=5
 951/951 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Security/DoctrineUserProvider.php                                                                                                                                         
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  80     Parameter $user of method App\Security\DoctrineUserProvider::upgradePassword() has invalid type Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface.  
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Security/KimaiUserProvider.php                                                                                                                                         
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  87     Parameter $user of method App\Security\KimaiUserProvider::upgradePassword() has invalid type Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface.  
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
                                                                                                                        
 [ERROR] Found 2 errors                                                                                                 

Script vendor/bin/phpstan analyse src -c phpstan.neon --level=5 handling the phpstan event returned with error code 1
```

Composer update:
```bash
$ composer update 
Loading composer repositories with package information                                                                                                                                                     Restricting packages listed in "symfony/symfony" to "4.4.*"
Updating dependencies
Lock file operations: 0 installs, 1 update, 0 removals
  - Upgrading phpstan/phpstan-symfony (1.2.23 => dev-PasswordAuthenticatedUserStubFilesExtension d879c1a)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 0 installs, 1 update, 0 removals
  - Downloading phpstan/phpstan-symfony (dev-PasswordAuthenticatedUserStubFilesExtension d879c1a)
  - Upgrading phpstan/phpstan-symfony (1.2.23 => dev-PasswordAuthenticatedUserStubFilesExtension d879c1a): Extracting archive
Package sensio/framework-extra-bundle is abandoned, you should avoid using it. Use Symfony instead.
Package symfony/debug is abandoned, you should avoid using it. Use symfony/error-handler instead.
Package symfony/inflector is abandoned, you should avoid using it. Use EnglishInflector from the String component instead.
Package php-cs-fixer/diff is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
composer/package-versions-deprecated: Generating version class...
composer/package-versions-deprecated: ...done generating version class
139 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```

Afterwards:
```bash
$ vendor/bin/phpstan analyse src -c phpstan.neon --level=5
 951/951 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
                                                                                                                        
 [OK] No errors                                                                                                         

```

--- a/extension.neon
+++ b/extension.neon
@@ -58,8 +58,6 @@ parameters:
 		- stubs/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.stub
 		- stubs/Symfony/Component/Security/Core/Authorization/Voter/Voter.stub
 		- stubs/Symfony/Component/Security/Core/Authorization/Voter/VoterInterface.stub
-		- stubs/Symfony/Component/Security/Core/User/PasswordAuthenticatedUserInterface.stub
-		- stubs/Symfony/Component/Security/Core/User/PasswordUpgraderInterface.stub
 		- stubs/Symfony/Component/Serializer/Encoder/ContextAwareDecoderInterface.stub
 		- stubs/Symfony/Component/Serializer/Encoder/DecoderInterface.stub
 		- stubs/Symfony/Component/Serializer/Encoder/EncoderInterface.stub
@@ -296,6 +294,10 @@ services:
 		tags: [phpstan.broker.dynamicMethodReturnTypeExtension]
 	-
 		class: PHPStan\Symfony\InputBagStubFilesExtension
+		tags:
+			- phpstan.stubFilesExtension
+	-
+		class: PHPStan\Symfony\PasswordAuthenticatedUserStubFilesExtension
 		tags:
 			- phpstan.stubFilesExtension
 

--- a/src/Symfony/InputBagStubFilesExtension.php
+++ b/src/Symfony/InputBagStubFilesExtension.php
@@ -2,15 +2,28 @@
 
 namespace PHPStan\Symfony;
 
+use PHPStan\BetterReflection\Reflector\Exception\IdentifierNotFound;
+use PHPStan\BetterReflection\Reflector\Reflector;
 use PHPStan\PhpDoc\StubFilesExtension;
-use function class_exists;
 
 class InputBagStubFilesExtension implements StubFilesExtension
 {
 
+	/** @var Reflector */
+	private $reflector;
+
+	public function __construct(
+		Reflector $reflector
+	)
+	{
+		$this->reflector = $reflector;
+	}
+
 	public function getFiles(): array
 	{
-		if (!class_exists('Symfony\Component\HttpFoundation\InputBag')) {
+		try {
+			$this->reflector->reflectClass('Symfony\Component\HttpFoundation\InputBag');
+		} catch (IdentifierNotFound $e) {
 			return [];
 		}
 

--- a/src/Symfony/PasswordAuthenticatedUserStubFilesExtension.php
+++ b/src/Symfony/PasswordAuthenticatedUserStubFilesExtension.php
@@ -2,15 +2,28 @@
 
 namespace PHPStan\Symfony;
 
+use PHPStan\BetterReflection\Reflector\Exception\IdentifierNotFound;
+use PHPStan\BetterReflection\Reflector\Reflector;
 use PHPStan\PhpDoc\StubFilesExtension;
-use function interface_exists;
 
 class PasswordAuthenticatedUserStubFilesExtension implements StubFilesExtension
 {
 
+	/** @var Reflector */
+	private $reflector;
+
+	public function __construct(
+		Reflector $reflector
+	)
+	{
+		$this->reflector = $reflector;
+	}
+
 	public function getFiles(): array
 	{
-		if (!interface_exists('Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface')) {
+		try {
+			$this->reflector->reflectClass('Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface');
+		} catch (IdentifierNotFound $e) {
 			return [];
 		}
 

--- a/src/Symfony/PasswordAuthenticatedUserStubFilesExtension.php
+++ b/src/Symfony/PasswordAuthenticatedUserStubFilesExtension.php
@@ -3,14 +3,14 @@
 namespace PHPStan\Symfony;
 
 use PHPStan\PhpDoc\StubFilesExtension;
-use function class_exists;
+use function interface_exists;
 
 class PasswordAuthenticatedUserStubFilesExtension implements StubFilesExtension
 {
 
 	public function getFiles(): array
 	{
-		if (!class_exists('Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface')) {
+		if (!interface_exists('Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface')) {
 			return [];
 		}
 

--- a/src/Symfony/PasswordAuthenticatedUserStubFilesExtension.php
+++ b/src/Symfony/PasswordAuthenticatedUserStubFilesExtension.php
@@ -1,0 +1,23 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Symfony;
+
+use PHPStan\PhpDoc\StubFilesExtension;
+use function class_exists;
+
+class PasswordAuthenticatedUserStubFilesExtension implements StubFilesExtension
+{
+
+	public function getFiles(): array
+	{
+		if (!class_exists('Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface')) {
+			return [];
+		}
+
+		return [
+			__DIR__ . '/../../stubs/Symfony/Component/Security/Core/User/PasswordAuthenticatedUserInterface.stub',
+			__DIR__ . '/../../stubs/Symfony/Component/Security/Core/User/PasswordUpgraderInterface.stub',
+		];
+	}
+
+}

